nav2_localization/include/nav2_localization/interfaces/matcher2d_base.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef NAV2_LOCALIZATION__MATCHER2D_BASE_HPP_
+#define NAV2_LOCALIZATION__MATCHER2D_BASE_HPP_
+
+#include "geometry_msgs/msg/pose.hpp"
+#include "sensor_msgs/msg/laser_scan.hpp"
+#include "nav_msgs/msg/occupancy_grid.hpp"
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+
+namespace nav2_localization
+{
+
+/**
+ * @class Matcher2d
+ * @brief Abstract interface for a 2D matcher for localization purposes to adhere to with pluginlib
+ */
+class Matcher2d
+{
+public:
+    Matcher2d(){}
+
+	using Ptr = std::shared_ptr<nav2_localization::Matcher2d>;
+
+    /**
+     * @brief Computes how likely is it to read a measurement given a position in a map
+     * @param scan 2D laser-like measurement
+     * @param pose Hypothesis on the robot's pose
+     * @return Probability of how likely is it to perceive the inputted measurement assuming that the robot is in the given pose inside the provided map
+     */
+	virtual float match(
```

Comment:
No corrected pose?

Reply:
Many matchers will match in a region, so it will try to match given an initial pose and a measurement and give you back out a refined pose and a score for that refined pose